2-AvailabilityCheck/Get-AvailabilityInformation.ps1

```diff
@@ -48,583 +48,626 @@
+                if (-not ($ConsolidatedSKUs[$size.name].Locations -contains $Region)) {
+                    $ConsolidatedSKUs[$size.name].Locations += $Region
+                }
+            }
+        }
+    }
+    # Convert the hash table to an array and save the VM SKUs to a JSON file
+    $SKUs = $ConsolidatedSKUs.Values
+    # Save VM SKUs to a JSON file
+    Out-JSONFile -Data $SKUs -fileName "Azure_SKUs_VM.json"
+    return @{
+    Data = $SKUs
+    }
+}
 
-# Find the Microsoft.Sql provider from the available providers
-Write-Output "  Retrieving SQL managed instance SKU regions information"
-$SQL_ManagedInstance_Provider = $Resources_All | Where-Object { $_.Namespace -ieq "Microsoft.Sql" }
-if ($SQL_ManagedInstance_Provider) {
-    # Select the resource type for SQL managed instance SKUs
-    $SQL_ManagedInstance_ResourceType = $SQL_ManagedInstance_Provider.ResourceTypes | Where-Object { $_.Type -ieq "managedInstances" }
-    if ($SQL_ManagedInstance_ResourceType) {
-        Write-Output "  Adding SQL managed instance SKUs for consolidated regions"
-        $SQL_ManagedInstance_Regions = $SQL_ManagedInstance_ResourceType.Locations
-        $TotalSQLManagedInstanceRegions = $SQL_ManagedInstance_Regions.Count
-        $CurrentSQLManagedInstanceRegionIndex = 0
-        foreach ($region in $SQL_ManagedInstance_Regions) {
-            $CurrentSQLManagedInstanceRegionIndex++
-            # Convert the display region into a region code for the URL
-            $regionCode = ($region -replace '\s','').ToLower()
-            Write-Output ("    Retrieving SQL managed instance SKU for region {0:D03} of {1:D03}: {2}" -f $CurrentSQLManagedInstanceRegionIndex, $TotalSQLManagedInstanceRegions, $region)
-            $SQL_ManagedInstance_Uri = "$REST_BaseUri/$REST_SubscriptionId/providers/Microsoft.Sql/locations/$regionCode/capabilities?api-version=2021-02-01-preview" 2>$null
-            try {
-                $SQL_ManagedInstance_Response = Invoke-RestMethod -Uri $SQL_ManagedInstance_Uri -Headers $REST_Headers -Method Get
-                # Select only the supportedManagedInstanceVersions property from the response
-                $FilteredCapabilities = $SQL_ManagedInstance_Response | Select-Object -Property supportedManagedInstanceVersions
-                # Rebuild supportedManagedInstanceVersions
-                if ($FilteredCapabilities -and $FilteredCapabilities.supportedManagedInstanceVersions) {
-                    $FilteredCapabilities.supportedManagedInstanceVersions = $FilteredCapabilities.supportedManagedInstanceVersions | ForEach-Object {
-                        $allSkus = @()
-                        if ($_.supportedEditions) {
-                            foreach ($edition in $_.supportedEditions) {
-                                if ($edition.supportedFamilies) {
-                                    $allSkus += $edition.supportedFamilies
+function Import-SKU-SQLManagedInstance {
```

Comment:
This function is very similar to the SQL database one, consider creating a single function with parameters to reduce code duplication